cirq-core/cirq/study/sweepable_test.py

```diff
@@ -147,3 +147,42 @@ def test_to_sweep_resolver_list(r_list_gen):
 def test_to_sweep_type_error():
     with pytest.raises(TypeError, match='Unexpected sweep'):
         cirq.to_sweep(5)
+
+
+@pytest.mark.parametrize(
+    'sweepables',
+    [
+        [{'a': sympy.Symbol('1 GHZ')}, {'a': sympy.Symbol('1.5 GHZ')}, {'b': 2}],
```

Comment:
Putting units in symbols like this seems like a bad idea because normally we consider symbols to be "unresolved" and thus subject to further resolution. In all of our internal code we use units as name suffixes so that we would do something like `a_GHz` instead, and then leave the values as plain numbers, which the parameter resolution code already treats as values that do not need further resolution. Have you considered a solution in this direction?

Reply:
+1 on Matthew said. Based on this idea, I think this direction can be done via modifying the `sweep_to_proto` and `sweep_from_proto` [here](https://github.com/quantumlib/Cirq/blob/v1.4.0/cirq-google/cirq_google/api/v2/sweeps.py#L24).

My thought is:
1. when calling `sweep_to_proto`, the sweeping name will suffix with a unit name if the sweeping value has unit.
2. when calling `sweep_from_proto`, check if the sweeping name has a unit suffix or not. If yes, remove the suffix and add the unit to the parameters. If not, same logic as before.

The caveat, of course, is we need to be careful about sweep parameter names. But note this approach will impact `cirq_google` only (instead of `cirq-core`) and we can also add a sanity check on the original name to make sure that original name doesn't contain a unit suffix. So this approach should be pretty safe IMO.

wdyt?

Reply:
I don't understand why a symbol should have a unit attached to it?! multiplying a sympy symbol with a unit raises a TypeError. so an object on its own can't be both a symbol and value with unit. 

if the context indicates that it does (e.g. `freq = sympy.Sybol('f'))` is assumed to be in `MHz` then that information lives somewhere else that is not the symbol or the parameter name, somewhere the symbol gets resolved to a float that gets the unit attached to it.

also importing cirq-google inside cirq-core isn't a good idea